dace/sdfg.py

```diff
@@ -413,17 +413,21 @@ def start_state(self):
             return source_nodes[0]
         if self._start_state is None:
             raise ValueError('Ambiguous or undefined starting state for SDFG')
+        if self._start_state not in source_nodes:
+            raise ValueError("Start state {} not found in source nodes".format(
+                self._start_state))
+        return self._start_state
```

Comment:
What if the graph is one state in a loop and an exit? The start state will not be a source node..

Reply:
The only way around that as far as I can tell is to allow _any_ state to be a start state. I doubt our codegen supports that?

Reply:
I think this is what _start_state is for - to allow a non-source-node to be a starting state

Reply:
Not necessarily, it's possible to have multiple source nodes, which is the situation where you'd need user input to decide where to start (not that this ever happens in practice, as far as I know).